test/Microsoft.Extensions.Localization.Tests/ResourceManagerStringLocalizerFactoryTest.cs

```diff
@@ -99,8 +99,8 @@ public void Create_FromNameLocation_ReturnsNewResultForDifferentLocation()
             var options = new Mock<IOptions<LocalizationOptions>>();
             options.Setup(o => o.Value).Returns(locOptions);
             var factory = new ResourceManagerStringLocalizerFactory(hostingEnvironment.Object, localizationOptions: options.Object);
-            var location1 = typeof(ResourceManagerStringLocalizer).GetTypeInfo().Assembly.FullName;
-            var location2 = typeof(ResourceManagerStringLocalizerFactoryTest).GetTypeInfo().Assembly.FullName;
+            var location1 = new AssemblyName(typeof(ResourceManagerStringLocalizer).GetTypeInfo().Assembly.FullName).Name;
+            var location2 = new AssemblyName(typeof(ResourceManagerStringLocalizerFactoryTest).GetTypeInfo().Assembly.FullName).Name;
```

Comment:
The test gave the impression that we should be passing in the full assemblyName to `Create(string, string)` which would make the ResourceManager look for something like `TestAssembly, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e.Resources.TestClass`. We actually want to pass the "short" assembly name there. But we didn't notice because the Assert in that test only cares if it produces Same/NotSame factories.

TL:DR: The semantics haven't changed, this was just wrong to start with.